src/main/java/Quokka.java

```diff
@@ -0,0 +1,244 @@
+                System.out.println("    " + (i + 1) + ". " + tasks.get(i));
+            }
+        }
+    }
+
+    private static void saveTasksToFile() {
+        try (PrintWriter writer = new PrintWriter(new FileWriter(DATA_FILE_PATH))) {
+            for (int i = 0; i < tasks.size(); i++) {
+                writer.println(tasks.get(i).toString());
+            }
+        } catch (IOException e) {
+            System.out.println("Error occurred while saving tasks to file: " + e.getMessage());
+        }
+    }
+
+    private static void loadTasksFromFile() {
+        File file = new File(DATA_FILE_PATH);
+        if (!file.exists()) {
+            System.out.println("No existing data file found. Starting with empty task list.");
+            return;
+        }
+
+        try (Scanner scanner = new Scanner(file)) {
+            while (scanner.hasNextLine() && tasks.size() < MAX_TASKS) {
+                String taskData = scanner.nextLine();
+                Task task = Task.parseFromFileString(taskData);
+                if (task != null) {
+                    tasks.add(task);
+                }
+            }
+        } catch (FileNotFoundException e) {
+            System.out.println("Data file not found: " + e.getMessage());
+        } catch (QuokkaException e) {
+            System.out.println("Data file contains corrupted data: " + e.getMessage());
+        }
+    }
+
+    private static void markTaskAsDone(String userInput) {
+        try {
+            updateTaskStatus(userInput, true, "Nice! I've marked this task as done:");
+        } catch (QuokkaException e) {
+            System.out.println("     Error: " + e.getMessage());
+        }
+    }
+
+    private static void markTaskAsNotDone(String userInput) {
+        try {
+            updateTaskStatus(userInput, false, "OK, I've marked this task as not done yet:");
+        } catch (QuokkaException e) {
+            System.out.println("     Error: " + e.getMessage());
+        }
+    }
+
+    private static void updateTaskStatus(String userInput, boolean newStatus, String statusMessage) {
+        try {
+            String[] parts = userInput.split(" ", 2);
+            if (parts.length == 2) {
+                int taskIndex = Integer.parseInt(parts[1]) - 1;
+                if (taskIndex >= 0 && taskIndex < tasks.size()) {
+                    tasks.get(taskIndex).setStatus(newStatus);
+                    System.out.println("    " + statusMessage);
+                    System.out.println("      " + tasks.get(taskIndex));
+                } else {
+                    throw new QuokkaException("Invalid task index.");
+                }
+            } else {
+                throw new QuokkaException("Please provide a valid task index to mark as done or not done.");
+            }
+        } catch (NumberFormatException e) {
+            System.out.println("     Error: Invalid task index format.");
+        }
+    }
+
+    public static void main(String[] args) {
```

Comment:
Avoid long methods. Take corrective action if the method exceeds 30 lines of code.